systemtest/src/test/java/io/strimzi/systemtest/operators/RecoveryST.java

```diff
@@ -228,6 +235,42 @@ void testRecoveryFromKafkaBridgeMetricsConfigDeletion() {
         timeMeasuringSystem.stopOperation(timeMeasuringSystem.getOperationID());
     }
 
+    @Test
+    void testRecoveryFromImpossibleMemoryRequest() {
```

Comment:
Is this test different from `io.strimzi.systemtest.rollingupdate.KafkaRollerST#testKafkaPodPending` apart from setting memory instead of CPU?

Reply:
Yes, I think so, as least potentially...

* That test stands up a cluster first and then updates it with an impossible request. One of the things we _ought_ to be checking there is that only one pod becomes unschedulable (i.e. we don't progressively take down the cluster) and that recovery is possible.
* This test starts with an impossible request (so all pods are unschedulable) and we want to ensure that recovery is impossible. My recollection was that this is how, despite us already having protection against the former in the `KafkaRoller`, we still had the recent bug with the latter.

Reply:
So assuming I'm correct about this, I guess we need to improve `KafkaRollerST#testKafkaPodPending` so that it waits for multiple reconciliations and verifies that N-1 of the pods are still running.

Reply:
Ok. Thanks for the explanation.

Reply:
I guess I can add these checks to all KafkaRoller tests which @stanlyDoge added in his KR PR -> not just to the `testKafkaPodPending`